bin/sync_slurm_tier1_projects.py

```diff
@@ -0,0 +1,174 @@
+            logging.info("processing section %s", section)
+            projects.append(ProjectIniConfig(
+                name=section,
+                group=section,
+                end_date=projects_config.get(section, "end_date"),
+                members=[m.strip() for m in projects_config.get(section, "members").split(",")],
+                cpu_hours=int(projects_config.get(section, "CPUhours", fallback=0)),
+                gpu_hours=int(projects_config.get(section, "GPUhours", fallback=0)),
+            ))
+        else:
+            projects.append(ProjectIniConfig(
+                name=section.replace("gpr_compute_", ""),
+                group=section,
+                end_date=projects_config.get(section, "end_date"),
+                members=[m.strip() for m in projects_config.get(section, "members").split(",")],
+    #            moderators=[m.strip() for m in projects_config.get(section, "moderators").split(",")],
+                cpu_hours=int(projects_config.get(section, "CPUhours", fallback=0)),
+                gpu_hours=int(projects_config.get(section, "GPUhours", fallback=0)),
+            ))
+
+    return projects
+
+class Tier1SlurmProjectSync(Sync):
+
+    CLI_OPTIONS = {
+        "clusters": (
+            "Cluster(s) (comma-separated) to sync for. ",
+            "strlist",
+            "store",
+            ["dodrio"],
+        ),
+        'project_ini': ('Ini file with projects information', str, 'store', None),
+    }
+
+    def update_project(self, project):
+        """
+        Retrieves project information from the AP -- if any -- and updates the projects' members.
+        """
+
+        try:
+            (_, group) = self.apc.group[project.group].get()
+        except HTTPError as _:
+            logging.error("Could not get project group %s data", project.group)
+            return project
+
+        logging.debug("Current group members: %s", project.members)
+        project = project._replace(members=group["members"])
+        logging.debug("Updated group members: %s", project.members)
+
+        return project
+
+    def do(self, dryrun):
+
+        projects = get_projects(self.options.project_ini)
+        # update project memberships from the AP if needed
+        projects = [self.update_project(p) for p in projects]
+
+        projects_members = [(set(p.members), p.name) for p in projects]  # TODO: verify enddates
+
+        # fetch slurm dbd information on accounts (projects), users and qos
+        slurm_account_info = get_slurm_acct_info(SyncTypes.accounts, exclude_accounts=VSC_EXCLUDE_ADMIN_GROUPS)
+        slurm_user_info = get_slurm_acct_info(SyncTypes.users)
+        slurm_qos_info = get_slurm_acct_info(SyncTypes.qos)
+
+        # The projects do not track active state of users, so we need to fetch all accounts as well
+        active_accounts = set([a["vsc_id"] for a in self.apc.account.get()[1] if a["isactive"]])
```

Comment:
use `get_accounts`?

Reply:
Yes, leftover from when the code was not based on Sync. Thanks.